build-binutils.py

```diff
@@ -0,0 +1,221 @@
+    """
+    host_mapping = {
+        "armv7l": "arm",
+        "ppc64le": "powerpc64le",
+        "ppc": "powerpc"
+    }
+    machine = platform.machine()
+    if machine in host_mapping.keys():
+        return host_mapping[machine]
+    else:
+        return machine
+
+
+def target_arch(target):
+    """
+    Returns the architecture from a target triple
+    :param target: Triple to deduce architecture from
+    :return: Architecture associated with given triple
+    """
+    return target.split("-")[0]
+
+
+def host_is_target(target):
+    """
+    Checks in the current target triple the same as the host?
+    :param target: Triple to match host architecture against
+    :return: True if host and target are same, False otherwise
+    """
+    return host_arch_target() == target_arch(target)
+
+
+def parse_parameters(root):
+    """
+    Parses parameters passed to the script into options
+    :param root: The directory where the script is being invoked from
+    :return: A 'Namespace' object with all the options parsed from supplied parameters
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-I",
+                        "--install-folder",
+                        help="""
+                        By default, the script will create a "build" folder in the same folder as this script
+                        and install binutils there. If you'd like to have it installed somewhere else, pass
+                        it to this parameter. This can either be an absolute or relative path.
+                        """,
+                        type=str,
+                        default=os.path.join(root.as_posix(), "build"))
+    parser.add_argument("-t",
+                        "--targets",
+                        help="""
+                        The script can build binutils targeting arm-linux-gnueabi, aarch64-linux-gnu,
+                        powerpc-linux-gnu, powerpc64le-linux-gnu, and x86_64-linux-gnu.
+
+                        You can either pass the full target or just the first part (arm, aarch64, x86_64, etc)
+                        or all if you want to build all targets (which is the default). It will only add the
+                        target prefix if it is not for the host architecture.
+                        """,
+                        nargs="+")
+    return parser.parse_args()
+
+
+def create_targets(targets):
+    """
+    Generate a list of targets that can be passed to the binutils compile function
+    :param targets: A list of targets to convert to binutils target triples
+    :return: A list of target triples
+    """
+    targets_dict = {
+        "arm": "arm-linux-gnueabi",
+        "aarch64": "aarch64-linux-gnu",
+        "powerpc64le": "powerpc64le-linux-gnu",
+        "powerpc": "powerpc-linux-gnu",
+        "x86_64": "x86_64-linux-gnu"
+    }
+    targets_list = []
+
+    if len(targets) == 1 and targets[0] == "all":
+        targets_list = list(targets_dict.values())
```

Comment:
Maybe simply `return list(targets_dict.values())` and un-indent the other branch (plus moving down `targets_list`).

Reply:
Thank you for the suggestion, done: https://github.com/ClangBuiltLinux/tc-build/commit/6c49b7a5bfc7950faf61dc140662046b7ebb6d3a

However, I realized that something like `./build-binutils.py -t x86_64 all` would error out so I fixed that here: https://github.com/ClangBuiltLinux/tc-build/commit/51670aafb24077903db226ef4b5c4fe150fa8925